File: app/src/main/java/com/boardgamegeek/model/PlaySaveResponse.java
package com.boardgamegeek.model;

import com.boardgamegeek.provider.BggContract;

import androidx.annotation.NonNull;
import okhttp3.OkHttpClient;
import okhttp3.Request;

public class PlaySaveResponse extends PlayPostResponse {
	private PlaySave playSave;

	public PlaySaveResponse(@NonNull OkHttpClient client, @NonNull Request request) {
		super(client, request);
	}

	@Override
	protected void saveContent(String content) {
		playSave = gson.fromJson(content, PlaySave.class);
		error = playSave.error;
	}

	public int getPlayCount() {
		if (hasError() || playSave == null) {
			return -1;
		} else {
			return playSave.getNumberOfPlays();
		}
	}

	public int getPlayId() {
		if (hasError() || playSave == null) {
			return BggContract.INVALID_ID;
		} else {
			return playSave.getPlayId();
		}
	}

	private static class PlaySave {
		@SuppressWarnings("unused") private int playid;
		@SuppressWarnings("unused") private int numplays;
		@SuppressWarnings("unused") private String html;
		@SuppressWarnings("unused") private String error;

		public int getPlayId() {
			return playid;
		}

		public int getNumberOfPlays() {
			return numplays;
		}
	}
}

File: app/src/main/java/com/boardgamegeek/service/PlaySaveResponse.kt
package com.boardgamegeek.service

import com.boardgamegeek.provider.BggContract
import okhttp3.OkHttpClient
import okhttp3.Request

class PlaySaveResponse(client: OkHttpClient, request: Request) : PlayPostResponse(client, request) {
    private var playSave: PlaySave? = null

    override fun saveContent(content: String?) {
        playSave = gson.fromJson(content, PlaySave::class.java)
        error = playSave?.error
    }

    val playCount: Int
        get() = if (hasError()) {
            -1
        } else {
            playSave?.numplays ?: -1
        }

    val playId: Int
        get() = if (hasError()) {
            BggContract.INVALID_ID
        } else {
            playSave?.playid ?: BggContract.INVALID_ID
        }

    @Suppress("SpellCheckingInspection")
    private class PlaySave {
        val playid = 0
        val numplays = 0
        @Suppress("unused")
        val html: String? = null
        val error: String? = null
    }
}